daemon/api_users.go

```diff
@@ -341,196 +313,33 @@ func createUser(c *Command, createData postUserCreateData) Response {
 		createData.Sudoer = true
 	}
 
-	var model *asserts.Model
-	var serial *asserts.Serial
-	createKnown := createData.Known
-	if createKnown {
-		var err error
-		st.Lock()
-		model, err = c.d.overlord.DeviceManager().Model()
-		st.Unlock()
-		if err != nil {
-			return InternalError("cannot create user: cannot get model assertion: %v", err)
-		}
-		st.Lock()
-		serial, err = c.d.overlord.DeviceManager().Serial()
-		st.Unlock()
-		if err != nil && !errors.Is(err, state.ErrNoState) {
-			return InternalError("cannot create user: cannot get serial: %v", err)
-		}
-	}
-
-	// special case: the user requested the creation of all known
-	// system-users
-	if createData.Email == "" && createKnown {
-		return createAllKnownSystemUsers(st, model, serial, &createData)
-	}
-	if createData.Email == "" {
-		return BadRequest("cannot create user: 'email' field is empty")
-	}
-
-	var username string
-	var opts *osutil.AddUserOptions
-	if createKnown {
-		username, opts, err = getUserDetailsFromAssertion(st, model, serial, createData.Email)
-	} else {
-		username, opts, err = getUserDetailsFromStore(storeFrom(c.d), createData.Email)
-	}
+	createdUsers, err := doCreateUser(st, createData)
 	if err != nil {
-		return BadRequest("%s", err)
-	}
-
-	// FIXME: duplicated code
-	opts.Sudoer = createData.Sudoer
-	opts.ExtraUsers = !release.OnClassic
-
-	if err := osutilAddUser(username, opts); err != nil {
-		return BadRequest("cannot create user %s: %s", username, err)
-	}
-
-	if err := setupLocalUser(c.d.overlord.State(), username, createData.Email); err != nil {
-		return InternalError("%s", err)
-	}
-
-	result := userResponseData{
-		Username: username,
-		SSHKeys:  opts.SSHKeys,
-	}
-
-	if createData.singleUserResultCompat {
-		// return a single userResponseData in this case
-		return SyncResponse(&result)
```

Comment:
we lost this bit of logic that was intentional and need to preserve

Reply:
Also (hopefully correctly) restored now